nimble_orchestration/components.py

```diff
@@ -168,3 +168,38 @@ def color(self):
         The color of this component.
         """
         return self._color
+
+
+class Shelf:
+    """
+    A class for all the orchestraion information relating to a shelf
+    """
+    def __init__(self,
+                 assembled_shelf: AssembledComponent,
+                 device: Device):
+        self._assembled_shelf = assembled_shelf
+        self._device = device
+
+    @property
+    def assembled_shelf(self):
+        """
+        Return the Object describing the assembled shelf (currently this in an empty
+        shelf in the correct location on the rack).
+        This is an AssembledComponent
+        """
+        return self._assembled_shelf
+
+    @property
+    def device(self):
+        """
+        Return the Device object for the networking component that sits on this shelf
```

Comment:
As long as there are dimensions for the devices in NocoDB, it will be easy to auto-generate device placeholders. We can add things like fillets and "Reference Only" inset text so that they are not just plain blocks.

Reply:
Yeah, adding in placeholders for devices and adding in screws is something we need to do soon.